src/monodroid/jni/monodroid-glue.cc

```diff
@@ -298,29 +300,42 @@ MonodroidRuntime::open_from_update_dir (MonoAssemblyName *aname, [[maybe_unused]
 
 	const char *culture = reinterpret_cast<const char*> (mono_assembly_name_get_culture (aname));
 	const char *name    = reinterpret_cast<const char*> (mono_assembly_name_get_name (aname));
-	char *pname_raw_ptr;
+	size_t culture_len;
 
-	if (culture != nullptr && *culture != '\0')
-		pname_raw_ptr = utils.path_combine (culture, name);
+	if (culture != nullptr)
+		culture_len = strlen (culture);
 	else
-		pname_raw_ptr = utils.strdup_new (name);
+		culture_len = 0;
+	size_t name_len = strlen (name);
+
+	static_local_string<SENSIBLE_PATH_MAX> pname (name_len + culture_len);
+	if (culture_len > 0)
+		pname.append (culture, culture_len);
+	pname.append (name, name_len);
 
-	simple_pointer_guard<char[]> pname (pname_raw_ptr);
+	constexpr char dll_extension[] = ".dll";
+	constexpr size_t dll_extension_len = sizeof(dll_extension) - 1;
 
-	constexpr const char *format_none = "%s" MONODROID_PATH_SEPARATOR "%s";
-	constexpr const char *format_dll  = "%s" MONODROID_PATH_SEPARATOR "%s.dll";
+	bool is_dll = utils.ends_with (name, dll_extension);
+	size_t file_name_len = pname.length () + 1;
+	if (!is_dll)
+		file_name_len += dll_extension_len;
 
 	for (uint32_t oi = 0; oi < AndroidSystem::MAX_OVERRIDES; ++oi) {
 		override_dir = androidSystem.get_override_dir (oi);
 		if (override_dir == nullptr || !utils.directory_exists (override_dir))
 			continue;
 
-		const char *format = utils.ends_with (name, ".dll") ? format_none : format_dll;
-		simple_pointer_guard<char, false> fullpath (utils.monodroid_strdup_printf (format, override_dir, pname.get ()));
+		size_t override_dir_len = strlen (override_dir);
+		static_local_string<SENSIBLE_PATH_MAX> fullpath (override_dir_len + file_name_len);
```

Comment:
Do we need to include the count for the terminating null here?

Reply:
This is to account for the path separator